Include/object.h

```diff
@@ -87,7 +87,7 @@ whose size is determined when the object is allocated.
     1, type },
 
 #define PyVarObject_HEAD_INIT(type, size)       \
-    { PyObject_HEAD_INIT(type) size },
+    { PyObject_HEAD_INIT(NULL) size },
```

Comment:
IMHO this is not correct, in particular when a type defined in C has a custom metaclass.

Furthermore this leads to possibly confusing code when the invocations of this macro are not changed: the code then suggests that a particular value for "type" is used while this is ignored.

IMHO This patch should just update the invocations of PyObject_HEAD_INIT in CPython (if there is a need for such a patch at all).  Note that PyType_Ready looks at a type flag to see if a type has been initialised, not at the value for ob_type.